odoo_test_helper/fake_model_loader.py

```diff
@@ -52,34 +52,31 @@ def tearDown(self):
             super().tearDown()
     """
 
-    _original_registry = None
-    _original_module2modules = None
-    _module_name = None
-
     def __init__(self, env, __module__):
         self.env = env
         self._module_name = self.env.registry["base"]._get_addon_name(__module__)
+        self._original_registry = None
+        self._new_models = []
 
     def backup_registry(self):
         self._original_registry = {}
-        self._original_module_to_models = {}
         for model_name, model in self.env.registry.models.items():
             self._original_registry[model_name] = {
                 "base": model.__bases__,
                 "_fields": model._fields.copy(),
                 "_inherit_children": OrderedSet(model._inherit_children._map.keys()),
                 "_inherits_children": set(model._inherits_children),
             }
-        for key in module_to_models:
-            self._original_module_to_models[key] = list(module_to_models[key])
 
     def update_registry(self, odoo_models):
         # Ensure that fake model are in your module
-        # If you test are re-using fake model form an other module
+        # If you test are re-using fake model from an other module
         # the following code will inject it like it was in your module
         for model in odoo_models:
             if model not in module_to_models[self._module_name]:
                 module_to_models[self._module_name].append(model)
+            if model not in self._original_registry:
```

Comment:
This is wrong. Indeed the "model" in module_to_models are the original python class that come from the .py file.
And in the registry it's the real odoo model (ex: "res.partner")
So the model will never be in the _original_registry

Reply:
ehm sorry, you are wrong :)
`module_to_models` already contains the fake one at this stage. It's pointless to clone that mapping. Just get rid of fake models when done.